pkg/machinery/config/types/v1alpha1/v1alpha1_provider_test.go

```diff
@@ -0,0 +1,93 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package v1alpha1_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/siderolabs/talos/pkg/machinery/config/types/v1alpha1"
+)
+
+func TestInstallDiskSelector(t *testing.T) {
+	t.Parallel()
+
+	for _, test := range []struct {
+		name string
+
+		selector v1alpha1.InstallDiskSelector
+
+		expected string
+	}{
+		{
+			name: "size",
+
+			selector: v1alpha1.InstallDiskSelector{
+				Size: &v1alpha1.InstallDiskSizeMatcher{
+					MatchData: v1alpha1.InstallDiskSizeMatchData{
+						Op:   "<=",
+						Size: 256 * 1024,
+					},
+				},
+			},
+
+			expected: `disk.size <= 262144u && !disk.readonly && !disk.cdrom`,
+		},
+		{
+			name: "size and type",
+
+			selector: v1alpha1.InstallDiskSelector{
+				Size: &v1alpha1.InstallDiskSizeMatcher{
+					MatchData: v1alpha1.InstallDiskSizeMatchData{
+						Size: 1024 * 1024,
+					},
+				},
+				Type: v1alpha1.InstallDiskType("nvme"),
+			},
+
+			expected: `disk.size == 1048576u && disk.transport == "nvme" && !disk.readonly && !disk.cdrom`,
+		},
+		{
+			name: "size and type and modalias",
+
+			selector: v1alpha1.InstallDiskSelector{
+				Size: &v1alpha1.InstallDiskSizeMatcher{
+					MatchData: v1alpha1.InstallDiskSizeMatchData{
+						Size: 1024 * 1024,
+					},
+				},
+				Type:     v1alpha1.InstallDiskType("hdd"),
+				Modalias: "pci:1234:5678*",
+			},
+
+			expected: `disk.size == 1048576u && glob("pci:1234:5678*", disk.modalias) && disk.rotational &&
+!disk.readonly && !disk.cdrom`,
+		},
+		{
+			name: "ssd",
+
+			selector: v1alpha1.InstallDiskSelector{
+				Type: v1alpha1.InstallDiskType("ssd"),
+			},
+
+			expected: `disk.transport != "" && !disk.readonly && !disk.cdrom`,
```

Comment:
`disk.transport != ""` wouldn't this match any type?

Reply:
I won't match disks with empty `transport`... but probably this is wrong, I should add `!rotational` here